### ISSUE TITLE
Sophia needs new wheels

### ISSUE THREAD

**Issue by DefProc** [DoESLiverpool/somebody-should#198]
the nylon wheels for the x-axis (the visible one) are worn past the limits of their adjustment, and we need to replace them. 

The carriage has a visible wobble when moving that has a noticeable effect on cut accuracy.

**Comment by DefProc** [comment]
@goatchurchprime cut some new ones on the CNC, and they're fitted and work. (faster that waiting for an email reply from the supplier!) Cut from a nylon chopping board. 

CAD models and drawings for the wheel are at: https://cad.onshape.com/documents/8018ed64b1a74c8691e24656

We need to record the G-code files @goatchurchprime was using, so they can be found next time we need some more!

**Comment by goatchurchprime** [comment]
From Patrick's design:

![sophiaguidewheels](https://cloud.githubusercontent.com/assets/677254/22065854/fe314354-dd81-11e6-99a5-e43ebc28e8c9.png)

**Comment by goatchurchprime** [comment]
Here is the python-generating G-code that does this:

[sophiaguidewheel.py.zip](https://github.com/DoESLiverpool/somebody-should/files/714627/sophiaguidewheel.py.zip)

`from math import sin, cos

toolrad = 2.0
def helix(r, z1, z0):
    print("G1Z%.3f" % z1)
    print("X%.3f" % r)
    z = z1
    while True:
        blast = (z == z0)
        print("G3X%.3fI%.3fJ0Z%.3f" % (-r, -r, z))
        z = max(z - 1.1, z0)
        print("G3X%.3fI%.3dJ0Z%.3f" % (r, r, z))
        if blast:
            break
        z = max(z - 1.1, z0)
    
print("g1z8f1500")
print("g1x0y0")
helix(7.9/2-toolrad, 10, 1.5)
helix(6.5/2-toolrad, 1.5, -1)
helix((12+7.9)/2/2, 10, 7.5)
helix(12/2+toolrad, 10, 0)
print("G1z10")
print("M2")
`

**Comment by goatchurchprime** [comment]
@DefProc the unfinished wheels are on your desk.

**Comment by amcewen** [comment]
Presumably this would stop the ghosting on etching that Sophia is doing at the moment?

What finishing needs doing to the wheels?  Are they still on your desk @DefProc?

**Comment by amcewen** [comment]
From the organisers meeting, any update on this @DefProc?

**Comment by DefProc** [comment]
I thought I'd put the checked wheels in the laser cutter cupboard (although I didn't find them when I had a quick look at the end of July). I know there is a set of seconds (mostly with small imperfections in one edge from the cutter) in the drawers on top of my desk (right side of desk, centre column). 

I haven't fitted any replacements this year.

**Comment by DefProc** [comment]
After confirming with @goatchurchprime, I've ordered a small sheet of 10mm thick delrin sheet to machine some new wheels from. Should be with us by Tuesday 10th October.

Using delrin over the previous material (nylon) should make the wheels easier to machine, and more hard wearing in use.

**Comment by goatchurchprime** [comment]
Code is now:

```python
from math import sin, cos, radians, degrees

SFM = 800
IPR = 0.02
nflutes = 2
toolrad = 3.0/2
dz = 0.95

speed=SFM*3.82/(2*toolrad/25.4)
feed=speed*IPR*nflutes*25.4
ztop = 11
ztab = 1.0
zclear = 15
tabwidth = 1.0
tabwidthdeg = degrees((toolrad*2+tabwidth)/(12/2.0 + toolrad))
tabwidthn = max(2, int(360/tabwidthdeg+1))
tabwidthn = 3*int(tabwidthn/3)
#assert False, (tabwidthdeg, tabwidthn)



def helix(r, z1, z0, cx, cy, n, ktabbing=0):
    print("G1Z%.3f" % z1)
    px, py = r, 0
    print("G1X%.3fY%.3f" % (px+cx, py+cy))
    print("G1Z%.3f" % z1)
    z = z1
    
    i = 0
    izl = 0
    while izl <= n or ((i%n) != 0):
        i += 1
        z = max(z - dz*1.0/n, z0)
        th = radians((i%n)*1.0/n*360)
        npx, npy = cos(th)*r, sin(th)*r
        if ktabbing and (((i%n)%ktabbing) == int(ktabbing/2)) and z<ztab:
            print("G1Z%.3f" % ztab)
            print("G3X%.3fY%.3fI%.3fJ%.3f" % (npx+cx, npy+cy, -px, -py))
            print("G1Z%.3f" % z)
        else:
            print("G3X%.3fY%.3fI%.3fJ%.3fZ%.3f" % (npx+cx, npy+cy, -px, -py, z))
        px, py = npx, npy
        if z == z0:
            izl += 1 

n = 6
def roun(cx, cy):
    print("G1Z%dS%dF%d" % (zclear, speed, feed))
    print("G1X%.3fY%.3f" % (cx, cy))
    
    print("G1z%dF%d"%(ztop, feed))
    #return

    ah = 0.3  # make thicker
    helix(toolrad-0.5, ztop, 7.5+ah, cx, cy, n)
    helix(7.9/2-toolrad, ztop, 7.5+ah, cx, cy, n)
    helix((12.05+7.9)*0.5/2, ztop, 7.5+ah, cx, cy, n)
    helix(12.05/2+toolrad, ztop, 7.5+ah, cx, cy, n)
    helix(12.05/2+toolrad+1, ztop, 7.5+ah, cx, cy, n)

    helix(7.9/2-toolrad, 7.5+ah, 1.5+ah, cx, cy, n)
    helix(6.5/2-toolrad, 1.5+0.2+ah, -0.2, cx, cy, n)
    
    helix(12.05/2+toolrad, 8.5+ah, -0.2, cx, cy, tabwidthn, int((tabwidthn+1)/3))


RD = (12/2+toolrad+2.2)*2
th0 = radians(20)
th1 = radians(20+60)

for i0 in range(4,5):
    i1 = 6-i0
    roun(0 - cos(th0)*RD*i0 - cos(th1)*RD*i1, 
         -15 - sin(th0)*RD*i0 - sin(th1)*RD*i1)

print("G1z%d"%zclear)
print("M2")
```

**Comment by DefProc** [comment]
These are in the laser parts cupboard.

<blockquote class="twitter-tweet" data-lang="en">
<p lang="en" dir="ltr">A fresh batch of CNC&#39;d, delrin, guide wheels for <a href="https://twitter.com/DoESLiverpool?ref_src=twsrc%5Etfw">@DoESLiverpool</a>&#39;s small laser cutter, thanks to <a href="https://twitter.com/goatchurch?ref_src=twsrc%5Etfw">@goatchurch</a> <a href="https://twitter.com/hashtag/weeknotes?src=hash&amp;ref_src=twsrc%5Etfw">#weeknotes</a> <a href="https://t.co/oJpDvw0Kyh">pic.twitter.com/oJpDvw0Kyh</a></p>

![image](https://user-images.githubusercontent.com/343000/31776023-33e31288-b4e2-11e7-9c7c-b1ad75917124.png)

&mdash; Def-Proc (@DefProc) <a href="https://twitter.com/DefProc/status/921019337015775232?ref_src=twsrc%5Etfw">October 19, 2017</a></blockquote>

**Comment by DefProc** [comment]
The wheel alignment is so sloppy at the moment, that bi-directional engraving has obvious ghosting. 

Although I'm worried that fixing anything around the laser carriage will upset the drive belt that is on its last legs, and we don't yet have a replacement for: #565

**Comment by johncain39** [comment]
Yes, myself and Darran did this and the belt about 3/4 weeks ago.
